packages/teams-js/src/public/location.ts

```diff
@@ -46,82 +44,107 @@ export namespace location {
-  /**
-   * @deprecated
-   * As of 2.0.0, please use {@link location.getLocation location.getLocation(props: LocationProps): Promise\<Location\>} instead.
-   * @param props {@link LocationProps} - Specifying how the location request is handled
-   * @param callback - Callback to invoke when current user location is fetched
-   */
-  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void;
-  export function getLocation(
-    props: LocationProps,
-    callback?: (error: SdkError, location: Location) => void,
-  ): Promise<Location> {
+  export function getCurrentLocation(): Promise<Location> {
     ensureInitialized(FrameContexts.content, FrameContexts.task);
+    return sendAndHandleError('location.getLocation', { allowChooseLocation: false, showMap: false });
+  }
 
-    return callCallbackWithErrorOrResultFromPromiseAndReturnPromise<Location>(getLocationHelper, callback, props);
+  export function hasPermission(): Promise<boolean> {
+    const permissions: DevicePermission = DevicePermission.GeoLocation;
+
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('permissions.has', permissions));
+    });
   }
 
-  function getLocationHelper(props: LocationProps): Promise<Location> {
-    return new Promise<Location>(resolve => {
-      if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
-        throw { errorCode: ErrorCode.OLD_PLATFORM };
-      }
-      if (!props) {
-        throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
-      }
-      if (!isSupported()) {
-        throw errorNotSupportedOnPlatform;
-      }
-      resolve(sendAndHandleError('location.getLocation', props));
+  export function requestPermission(): Promise<boolean> {
+    const permissions: DevicePermission = DevicePermission.GeoLocation;
+
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('permissions.request', permissions));
     });
   }
 
+  export function isSupported(): boolean {
+    return runtime.supports.location ? true : false;
+  }
+
   /**
-   * Shows the location on map corresponding to the given coordinates
+   * @deprecated
+   * As of 2.0.0-beta.4, please use one of the following functions:
+   * - {@link location.getCurrentLocation location.getCurrentLocation(): Promise\<Location\>}
+   * - {@link location.map.chooseLocation location.map.chooseLocation(): Promise\<Location\>}
    *
-   * @param location {@link Location} - which needs to be shown on map
-   * @returns Promise that will be fulfilled when the operation has completed
+   * @param props {@link LocationProps} - Specifying how the location request is handled
+   * @param callback - Callback to invoke when current user location is fetched
    */
-  export function showLocation(location: Location): Promise<void>;
+  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void {
+    if (!callback) {
```

Comment:
Is this preserving the old behavior? I don't see this error being called anywhere, and we should make sure that we aren't altering how this function works (from the caller's perspective) as part of this change.

Reply:
Seems like callback was required before and is required now. The function was overloaded to convert it to promises. I will dm you to understand what I am missing

Reply:
@TrevorJoelHarris @lakhveerk I believe the confusion for this stems from the V1 `getLocation` and the old V2.0.0 `getLocation`.  V1's `getLocation` had a required `callback`, while V2.0.0 made this callback optional and would return a `Promise<void>`.  

I believe Trevor you are looking at the old code for `getLocation` that did overload the result into being a promise if a callback wasn't provided, but was effectively backwards compatible in functionality. i.e if callback was provided it would execute as the old `getLocation`.  Meanwhile Lakhveer converted the old backwards compatible function stub to being the implementation instead since it appears that `getLocation` is no longer needed and thus the overload logic wouldn't be needed anymore.

I believe for now it might be best to convert it back to the V2.0.0 behavior of overloading to promises with an optional callback handler for BC routing and then deprecate that functionality at a later time to what Lakhveer has here, but would be ok with discussing this later with you two. If we do proceed with making this specific change, I think it would be a good idea to communicate this out, in case someone is reliant on the current functionality.